### PR TITLE
Issue/68

### PR DESCRIPTION
**Pull request by loganasherjones**
This resolve beer-garden/beer-garden#68

We have initial support for prometheus metrics!

This will be running on `2338` by default. I'm open to different suggestions.

We should now have endpoints for the following:

1. Overall HTTP Server Metrics
  * This is in the form of a prometheus histogram, so we have
    latency, counts, and sums for all the endpoints and each
    method on those endpoints

2. In Progress/Waiting Request Gauges
  * These gauges tell us how many requests are IN_PROGRESS
    and how many requests have been created but not serviced.

### CODE REVIEW

File: brew_view/base_handler.py [hunk 1]
```diff
@@ -41,8 +55,22 @@ def set_default_headers(self):
...
+    def prometheus_endpoint(self):
+        """Removes Mongo ID from endpoint."""
+        to_return = self.request.path
+        for mongo_id in re.findall(self.MONGO_ID_PATTERN, self.request.path):
+            to_return = to_return.replace(mongo_id, '<ID>')
+        return to_return
+
     def prepare(self):
         """Called before each verb handler"""
+        self.request.headers['created_time_ms'] = self.current_time_millis
```
Comment (hazmat345) [inline brew_view/base_handler.py]: Does this need to be on a header? Looks like it just gets deleted later...

File: brew_view/controllers/request_api.py [hunk 2]
```diff
@@ -41,6 +54,27 @@ def get(self, request_id):
 
         self.write(self.parser.serialize_request(req, to_string=False))
 
+    def _update_completed_request_metrics(self, request):
+        latency = (
+            (datetime.datetime.utcnow() - request.updated_at).total_seconds() * 1000
```
Comment (hazmat345) [inline brew_view/controllers/request_api.py]: Does this give us actual millisecond precision?

Comment (loganasherjones) [reply brew_view/controllers/request_api.py]: Yes

File: brew_view/base_handler.py (hunk 1 above)
Comment (loganasherjones) [reply brew_view/base_handler.py]: Yeah, I did not know I can just add things to the `self.request`. I will do that tonight.

**Pull request merged**